datadog_checks_dev/datadog_checks/dev/tooling/commands/dep.py

```diff
@@ -124,207 +88,159 @@ def freeze():
-            # look for the dependency with the same marker and name since the version can be different
-            if check_dependency_definition.same_name_marker(agent_dependency_definition)
-            and version != agent_dep_version
-        }
-
-        if deps_to_update:
-            files_updated += 1
-            for dependency_definition, new_version in deps_to_update.items():
-                dependency_definition.requirement.specifier = new_version
-
-            if has_project_file(check_name):
-                update_check_dependencies(check_name, list(deps_to_update))
-            else:
-                check_req_file = get_check_req_file(check_name)
-                old_lines = read_file_lines(check_req_file)
-                new_lines = old_lines.copy()
+    if check_errors:
+        for error in check_errors:
+            echo_failure(error)
+        abort()
 
-                for dependency_definition, new_version in deps_to_update.items():
-                    dependency_definition.requirement.specifier = new_version
-                    new_lines[dependency_definition.line_number] = f'{dependency_definition.requirement}\n'
+    updated_checks = set()
+    for name, python_versions in check_dependencies.items():
+        check_dependency_definitions = get_dependency_set(python_versions)
+        agent_dependency_definitions = get_dependency_set(agent_dependencies[name])
 
-                write_file_lines(check_req_file, new_lines)
+        if check_dependency_definitions != agent_dependency_definitions:
+            for dependency_definition in agent_dependency_definitions:
+                updated_checks.update(update_project_dependency(python_versions, dependency_definition))
 
-    if not files_updated:
+    if not updated_checks:
         echo_info('All dependencies synced.')
+        return
+
+    for check_name in sorted(updated_checks):
+        update_check_dependencies(check_name, check_dependencies)
 
-    echo_info(f'Files updated: {files_updated}')
+    echo_info(f'Files updated: {len(updated_checks)}')
 
 
 async def get_version_data(url):
     async with request('GET', url) as response:
         try:
-            info = orjson.loads(await response.read())['info']
+            data = orjson.loads(await response.read())
         except Exception as e:
             raise type(e)(f'Error processing URL {url}: {e}')
         else:
-            return (
-                info['name'],
-                info['version'],
-                {
-                    extract_classifier_value(c)
-                    for c in info['classifiers']
-                    if c.startswith('Programming Language :: Python ::')
-                },
-            )
+            return data['info']['name'], data['releases']
 
 
 async def scrape_version_data(urls):
-    package_data = defaultdict(lambda: {'version': '', 'classifiers': set()})
+    package_data = {}
 
     async with Pool() as pool:
```

Comment:
https://pypi.org/project/python-binary-memcached/  and https://pypi.org/project/mmh3/ are not cooperating, `python-binary-memcached` updates as:
```
python-binary-memcached==0.31.1; sys_platform != 'win32' and python_version > '3.0'
python-binary-memcached==none; sys_platform != 'win32' and python_version < '3.0'
```

Reply:
Nice catch, fixed!